### ISSUE TITLE
7.6.2 "TypeError: set() got an unexpected keyword argument 'nx'"

### ISSUE THREAD

**Issue by colincoghill** [getsentry/sentry#1816]
Using redis, the worker fails to start, reporting:

```
   File "/home/sentry/ve/local/lib/python2.7/site-packages/sentry/buffer/redis.py", line 92, in process_pending
      if not self.conn.set(lock_key, '1', nx=True, ex=60):
   File "/home/sentry/ve/local/lib/python2.7/site-packages/sentry/../nydus/db/base.py", line 132, in __call__
     return self.__cluster.execute(self.__path, args, kwargs)
   File "/home/sentry/ve/local/lib/python2.7/site-packages/sentry/../nydus/db/base.py", line 78, in execute
      results.append(func(*args, **kwargs))
   TypeError: set() got an unexpected keyword argument 'nx' 
```

Applying https://github.com/getsentry/sentry/commit/14315e7a5de0d28021ace90baadeeabe44a34ebc appears to fix the problem. Presumably this needs to be rolled into 7.6, 7.7 ?

**Comment by dcramer** [comment]
I believe you're just running an unsupported old version of Redis. We require 2.6.12 iirc

**Comment by colincoghill** [comment]
We have 

sentry 7.6.2
redis-server 2.8.4
python-redis 2.7.2
nydus 0.11.0

Any suggestions for digging further?

**Comment by dcramer** [comment]
@colincoghill are you able to upgrade to the latest 7.x series release? It's also possible we fixed it later in versions and didnt backport. 

It's *also* possible that 2.8.4 doesnt include the changes. They backported it in some versions, but specifically the docs say 3.0.2 or newer. If its not in all we probably should update the requirements.

**Comment by colincoghill** [comment]
I tried 7.7.0, with the same problem. 

Ok, redis 3.0.3 works like a charm. 

Thank you!

( I was only using 2.8.4 because it's what came with Ubuntu Trusty)

**Comment by dcramer** [comment]
Pushed an update to suggest 3.0.2 or newer since its safer. I have a feeling the change was only backported to 2.6.12 and not other versions.

**Comment by jaysoffian** [comment]
Huh, the official redis docs say it's been available since 2.6.12 - http://redis.io/commands/set
